File: Model/Result.vb
<Serializable()> _
Public Class Result

    Public SetResults(-1) As SetResult

    Public Sub AddSetResult(setResult As SetResult)
        If setResult Is Nothing Then Throw New ArgumentNullException("setResultRow")

        Dim goalCount1, goalCount2 As Integer

        If IsNumeric(setResult.Home) Then goalCount1 = Integer.Parse(setResult.Home)
        If IsNumeric(setResult.Guest) Then goalCount2 = Integer.Parse(setResult.Guest)

        ReDim Preserve Me.SetResults(SetResults.Length)
        Me.SetResults(SetResults.Length - 1) = New SetResult()
        Me.SetResults(SetResults.Length - 1).Home = goalCount1
        Me.SetResults(SetResults.Length - 1).Guest = goalCount2
    End Sub

    Public Sub Clear()
        ReDim Preserve Me.SetResults(-1)
    End Sub

    Public Overrides Function ToString() As String
        If SetResults.Length = 0 Then Return ""

        Dim resultString = New Text.StringBuilder()

        For Each setResult In SetResults
            resultString.AppendFormat("/{0}", setResult)
        Next setResult

        Return resultString.ToString.Substring(1)
    End Function

End Class

File: Model/Result.cs
using System;
using Microsoft.VisualBasic;

namespace POFF.Kicker.Model
{
    [Serializable()]
    public class Result
    {

        public SetResult[] SetResults = new SetResult[0];

        public void AddSetResult(SetResult setResult)
        {
            if (setResult is null)
                throw new ArgumentNullException("setResultRow");

            int goalCount1 = default, goalCount2 = default;

            if (Information.IsNumeric(setResult.Home))
                goalCount1 = int.Parse(setResult.Home.ToString());
            if (Information.IsNumeric(setResult.Guest))
                goalCount2 = int.Parse(setResult.Guest.ToString());

            Array.Resize(ref SetResults, SetResults.Length + 1);
            SetResults[SetResults.Length - 1] = new SetResult();
            SetResults[SetResults.Length - 1].Home = goalCount1;
            SetResults[SetResults.Length - 1].Guest = goalCount2;
        }

        public void Clear()
        {
            Array.Resize(ref SetResults, 0);
        }

        public override string ToString()
        {
            if (SetResults.Length == 0)
                return "";

            var resultString = new System.Text.StringBuilder();

            foreach (var setResult in SetResults)
                resultString.AppendFormat("/{0}", setResult);

            return resultString.ToString().Substring(1);
        }

    }
}